client/src/test/omtLinkProvider.test.ts

```diff
@@ -1,60 +1,40 @@
+import { stub } from 'sinon';
+import { DocumentLinkRequest, LanguageClient } from 'vscode-languageclient';
+
+type Case = { i: number, line: number, start: number, end: number, target: string | undefined, should: string };
 
 describe('OMTLinkProvider', () => {
     const docUri = getDocUri('one/imports.omt');
-    const expectedCount = 6;
     let actualDocumentLinks: DocumentLink[];
+    let sendRequestStub;
 
     before(async () => {
+        sendRequestStub = stub(LanguageClient.prototype, 'sendRequest')
+            .returns(Promise.resolve([
+                new DocumentLink(toRange(1, 4, 1, 19))
+            ]));
+
         await activate(docUri);
-        actualDocumentLinks = (await commands.executeCommand(
+        const temp = (await commands.executeCommand(
             'vscode.executeLinkProvider',
             docUri,
-        )) as DocumentLink[];
+        ));
+        actualDocumentLinks = temp as DocumentLink[];
     });
 
-    it('should find all links', () => {
-        expect(actualDocumentLinks.length).to.equal(expectedCount);
-    })
-
-    describe('uri imports', () => {
-        it('should make a link for a relative path', () => {
-            testLinkProvider(0, { range: toRange(3, 3, 3, 18), target: toUri('/relative.omt') });
-        });
-
-        it('should recognize a shorthand from the tsconfig in current project folder', () => {
-            testLinkProvider(1, { range: toRange(5, 4, 5, 24), target: toUri('/one/shorthanded.omt') });
-        });
-
-        it('should recognize a shorthand from the tsconfig in a parent project', () => {
-            testLinkProvider(5, { range: toRange(13, 4, 13, 27), target: toUri('/three/parentPathed.omt') });
-        });
-
-        it('should not ignore a shorthand from a sibling path', () => {
-            // without knowledge of how the project makes path shorthands we cannot determine if something starts with that.
-            // therefore it will become a link with an invalid uri and we let the IDE handle that
-            // ToDo, determine if it is wise to check if the shorthand is present in other configs and disable the link and give a warning with the language server
-            testLinkProvider(4, { range: toRange(11, 4, 11, 31), target: toUri('/one/@two/siblingShorthanded.omt') });
-        });
-
-        it('should ignore quotes', () => {
-            testLinkProvider(2, { range: toRange(7, 5, 7, 29), target: toUri('/one/quotedShorthand.omt') });
-            testLinkProvider(3, { range: toRange(9, 5, 9, 35), target: toUri('/one/doubleQuotedShorthand.omt') });
-        });
+    it('should request documentlinks from the server', () => {
+        // only test if the omtLinkProvider calls the right server functions
+        // and if it is called when the document is loaded
+        // the making of the links should be tested for the server
+        // and the rendering of the links is the responsibility of vscode
+        sendRequestStub.calledWith(
```

Comment:
`spy.calledWith()` is a function that returns a boolean, it is not an expectation in itself. Use `expect()` here.

Reply:
nice catch.
I will be using the assert function from sinon though.

the test failed so I will make a task for that:

- [ ] Fix it `should request documentlinks from the server` test